x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -267,6 +267,10 @@ protected void masterOperation(
                             if (sourceIndexMetadata.getNumberOfReplicas() > 0) {
                                 settings.put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, sourceIndexMetadata.getNumberOfReplicas());
                             }
+                            // Setting index.hidden has been initially set to true. We revert this to the value of the source index
+                            if (sourceIndexMetadata.isHidden() == false) {
+                                settings.put(IndexMetadata.SETTING_INDEX_HIDDEN, false);
+                            }
```

Comment:
There's a small catch here. If the setting `index.hidden` was set at all in the source index, the default value is `false`. Here we set the value explicitly to `false`. Hopefully, this does not cause any problems.

Reply:
Could we do something like 

```java
if (sourceIndexMetadata.isHidden() == false && sourceIndexMetadata.getSettings().keySet().contains(IndexMetadata.SETTING_INDEX_HIDDEN)) {
    settings.put(IndexMetadata.SETTING_INDEX_HIDDEN, false);
}
```

To only set the setting if it does indeed exist on the source as explicitly set?

Reply:
When the rollup index is created setting `index.hidden` is set to true. So, after rollup ends we want to revert this value.

I changed the code so that it removes setting (calling `setNull()`) if it is not set at all in the source index.

Reply:
Sounds good to me :+1: